# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1238106996 14400
# Node ID 643a96aec996a6da08e0c2850264872a4a98195d
# Parent 7c21256e35986a8f7ddcf843eeee44dd4a08d272
Fixed --with-ldap=plugin and --with-gssapi=plugin

--- a/src/auth/Makefile.am	Thu Mar 26 18:17:09 2009 -0400
+++ b/src/auth/Makefile.am	Thu Mar 26 18:36:36 2009 -0400
@@ -48,14 +48,6 @@ dovecot_auth_LDADD = \
 
 ldap_sources = db-ldap.c passdb-ldap.c userdb-ldap.c
 
-if ! LDAP_PLUGIN
-builtin_ldap_sources = $(ldap_sources)
-endif
-
-if ! GSSAPI_PLUGIN
-builtin_gssapi_sources = mech-gssapi.c
-endif
-
 dovecot_auth_SOURCES = \
 	auth.c \
 	auth-cache.c \
@@ -77,6 +69,7 @@ dovecot_auth_SOURCES = \
 	mech-login.c \
 	mech-cram-md5.c \
 	mech-digest-md5.c \
+	mech-gssapi.c \
 	mech-ntlm.c \
 	mech-otp.c \
 	mech-skey.c \
@@ -107,8 +100,7 @@ dovecot_auth_SOURCES = \
 	userdb-static.c \
 	userdb-vpopmail.c \
 	userdb-sql.c \
-	$(builtin_gssapi_sources) \
-	$(builtin_ldap_sources)
+	$(ldap_sources)
 
 headers = \
 	auth.h \
@@ -144,14 +136,14 @@ if GSSAPI_PLUGIN
 if GSSAPI_PLUGIN
 libmech_gssapi_la_LDFLAGS = -module -avoid-version
 libmech_gssapi_la_LIBADD = $(KRB5_LIBS)
-libmech_gssapi_la_CPPFLAGS = $(AM_CPPFLAGS) $(KRB5_CFLAGS)
+libmech_gssapi_la_CPPFLAGS = $(AM_CPPFLAGS) $(KRB5_CFLAGS) -DPLUGIN_BUILD
 libmech_gssapi_la_SOURCES = mech-gssapi.c
 endif
 
 if LDAP_PLUGIN
 libauthdb_ldap_la_LDFLAGS = -module -avoid-version
 libauthdb_ldap_la_LIBADD = $(LDAP_LIBS)
-libauthdb_ldap_la_CPPFLAGS = $(AM_CPPFLAGS)
+libauthdb_ldap_la_CPPFLAGS = $(AM_CPPFLAGS) -DPLUGIN_BUILD
 libauthdb_ldap_la_SOURCES = $(ldap_sources)
 endif
 
--- a/src/auth/db-ldap.c	Thu Mar 26 18:17:09 2009 -0400
+++ b/src/auth/db-ldap.c	Thu Mar 26 18:36:36 2009 -0400
@@ -2,7 +2,7 @@
 
 #include "common.h"
 
-#if defined(PASSDB_LDAP) || defined(USERDB_LDAP)
+#if defined(BUILTIN_LDAP) || defined(PLUGIN_BUILD)
 
 #include "network.h"
 #include "ioloop.h"
--- a/src/auth/mech-gssapi.c	Thu Mar 26 18:17:09 2009 -0400
+++ b/src/auth/mech-gssapi.c	Thu Mar 26 18:36:36 2009 -0400
@@ -23,7 +23,7 @@
 
 #include <stdlib.h>
 
-#ifdef HAVE_GSSAPI
+#if defined(BUILTIN_GSSAPI) || defined(PLUGIN_BUILD)
 
 #ifndef HAVE___GSS_USEROK
 #  define USE_KRB5_USEROK
--- a/src/auth/passdb-ldap.c	Thu Mar 26 18:17:09 2009 -0400
+++ b/src/auth/passdb-ldap.c	Thu Mar 26 18:36:36 2009 -0400
@@ -3,7 +3,7 @@
 #include "common.h"
 #include "passdb.h"
 
-#ifdef PASSDB_LDAP
+#if defined(PASSDB_LDAP) && (defined(BUILTIN_LDAP) || defined(PLUGIN_BUILD))
 
 #include "ioloop.h"
 #include "hash.h"
--- a/src/auth/userdb-ldap.c	Thu Mar 26 18:17:09 2009 -0400
+++ b/src/auth/userdb-ldap.c	Thu Mar 26 18:36:36 2009 -0400
@@ -3,7 +3,7 @@
 #include "common.h"
 #include "userdb.h"
 
-#ifdef USERDB_LDAP
+#if defined(USERDB_LDAP) && (defined(BUILTIN_LDAP) || defined(PLUGIN_BUILD))
 
 #include "hash.h"
 #include "str.h"

